tests/snmp/test_snmp_queue.py

```diff
@@ -26,25 +28,37 @@ def test_snmp_queues(duthosts, enum_rand_one_per_hwsku_hostname, localhost, cred
             q_keys.extend(q_keys_ns)
         if config_facts_ns and 'port_name_to_alias_map' in config_facts_ns:
             port_name_to_alias_map.update(config_facts_ns['port_name_to_alias_map'])
+            for port_name in config_facts_ns['port_name_to_alias_map']:
+                port_name_to_ns[port_name] = namespace
 
     if not q_keys:
         pytest.skip("No queues configured on interfaces")
 
     # Get alias : port_name map
     alias_port_name_map = {k: v for v, k in port_name_to_alias_map.items()}
 
-    q_interfaces = set()
+    q_interfaces = dict()  # {intf_name : set(queue_indexes)}
     # get interfaces which has configured queues
     for key in q_keys:
+        intf_idx = 0
+        queue_idx = 0
         intf = key.split('|')
         # 'QUEUE|Ethernet*|2'
         if len(intf) == 3:
-            q_interfaces.add(intf[1])
+            intf_idx = 1
+            queue_idx = 2
         # Voq chassis 'QUEUE|<hostname>|<asic_ns>|Ethernet*|2'
         elif len(intf) == 5:
             # Choose only interfaces on current linecard.
             if intf[1] == duthost.hostname:
-                q_interfaces.add(intf[3])
+                intf_idx = 3
+                queue_idx = 4
+        if intf_idx != 0:
+            if intf[intf_idx] in q_interfaces:
+                q_interfaces[intf[intf_idx]].add(intf[queue_idx])
```

Comment:
>add [](http://example.com/codeflow?start=45&length=3)

This line is duplicated in both if/else. To simplify, you just need to initialize the set() if it is empty.

Reply:
Updated as suggested